### ISSUE TITLE
Error while ingesting domains.registrations

### ISSUE THREAD

**Issue by xino12** [cloudquery/cq-provider-gcp#37]
When I try to ingest GCP dns registration I get the following error:

```
1:27PM ERR received resolve resources error error="googleapi: got HTTP response code 404 with body: <!DOCTYPE html>\n<html lang=en>\n  <meta charset=utf-8>\n  <meta name=viewport content=\"initial-scale=1, minimum-scale=1, width=device-width\">\n  <title>Error 404 (Not Found)!!1</title>\n  <style>\n    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}\n  </style>\n  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>\n  <p><b>404.</b> <ins>That’s an error.</ins>\n  <p>The requested URL <code>/v1beta1/projects/support-jx-general/location/-/registrations?alt=json&amp;pageToken=&amp;prettyPrint=false</code> was not found on this server.  <ins>That’s all we know.</ins>\n" 
```

Do you know what might be the error here? Is this supported? I can't find if it is now but I think it was supported at some point?

Thanks

**Comment by zagronitay** [comment]
Hi @xino12 ,
Thanks for reporting the issue.

We found a typo in the requested url.
It should be fixed on version 0.3.6

**Comment by xino12** [comment]
Hi @zagronitay so I tried ingesting domains.registrations and it seems it empty. Maybe I choose the wrong resource.

Anyway I tried adding dns to see if that ingested Cloud DNS domains and got this error:

```
8:27AM DBG panic: runtime error: invalid memory address or nil pointer dereference
8:27AM DBG [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1330c36]
8:27AM DBG 
8:27AM DBG goroutine 25036 [running]:
8:27AM DBG github.com/cloudquery/cq-provider-gcp/resources.resolveDomainsRegistrationGoogleDomainsDNSDsRecords(0x19a9bc0, 0xc002f2b380, 0x198dde0, 0xc004a4d7c0, 0xc007d9f180, 0x17d8234, 0x1d, 0xc, 0x180f182, 0x9a, ...)
8:27AM DBG 	/home/runner/work/cq-provider-gcp/cq-provider-gcp/resources/domains_registration.go:434 +0x56
8:27AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveColumns(0xc000454680, 0x19a9e80, 0xc00000e950, 0x19c8500, 0xc000454a80, 0x0, 0x19a9bc0, 0xc002f2b380, 0x198dde0, 0xc004a4d7c0, ...)
8:27AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.2.7/provider/schema/execution.go:157 +0x2a3
8:27AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveResourceValues(0xc000454680, 0x19a9e80, 0xc00000e950, 0x19c8500, 0xc000454a80, 0x0, 0x19a9bc0, 0xc002f2b380, 0x198dde0, 0xc004a4d7c0, ...)
8:27AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.2.7/provider/schema/execution.go:140 +0xba
8:27AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.resolveResources(0xc000454680, 0x19a9e80, 0xc00000e950, 0x19c8500, 0xc000454a80, 0x0, 0x19a9bc0, 0xc002f2b380, 0x198dde0, 0xc004a4d7c0, ...)
8:27AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.2.7/provider/schema/execution.go:114 +0x1df
8:27AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.callTableResolve(0xc000454680, 0x19a9e80, 0xc00000e950, 0x19c8500, 0xc000454a80, 0x0, 0x19a9bc0, 0xc002f2b380, 0x198dde0, 0xc004a4d7c0, ...)
8:27AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.2.7/provider/schema/execution.go:93 +0x232
8:27AM DBG github.com/cloudquery/cq-provider-sdk/provider/schema.ExecutionData.ResolveTable.func1(0x1, 0x0)
8:27AM DBG 	/home/runner/go/pkg/mod/github.com/cloudquery/cq-provider-sdk@v0.2.7/provider/schema/execution.go:51 +0xc5
8:27AM DBG golang.org/x/sync/errgroup.(*Group).Go.func1(0xc001689bc0, 0xc005fb50e0)
8:27AM DBG 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x59
8:27AM DBG created by golang.org/x/sync/errgroup.(*Group).Go
8:27AM DBG 	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x66
```

I can open another issue if you want.

**Comment by roneli** [comment]
Found the issue, will release a fix soon with new version of GCP (v0.4.0, has a few breaking changes), so we will also release this fix also with previous GCP version 0.3.X.

**Comment by roneli** [comment]
@xino12 released v0.3.7 with fix for the seg fault you received.
